File: src/main/java/br/com/model/entities/classes/Telefone/Telefone.java
package br.com.model.entities.classes.telefone;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
public class Telefone {
    @Id
    @GeneratedValue
    private Integer id;

    private String numero;

    public Telefone(String numero) {
        setNumero(numero);
    }

    public Telefone(Telefone telefone) {
        setId(telefone.getId());
        setNumero(telefone.getNumero());
    }

    public Telefone() {
    }

    @Override
    public String toString() {
        return "\n\n-- Telefone --\nID: " + getId() + "\nNúmero: " + getNumero();
    }

    // Getters / Setters

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNumero() {
        return numero;
    }

    public void setNumero(String numero) {
        this.numero = numero;
    }

}

File: src/main/java/br/com/model/entities/classes/Telefone.kt
package br.com.model.entities.classes

import br.com.model.entities.classes.usuario.Usuario
import jakarta.persistence.*

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
data class Telefone(
    @field:ManyToOne @field:JoinColumn(name = "id_usuario") private val usuario: Usuario,
    var numero: String
) {
    @field:Id
    @field:GeneratedValue
    var id: Int? = null
    override fun toString(): String {
        return "-- Telefone --\nID: $id\nNúmero: $numero\nUsuário: ${usuario.nome}"
    }
}
